### DISCUSSION TITLE
sensors for playlist + card

### DISCUSSION THREAD

**Discussion by jtbgroup** [boralyl/kodi-recently-added#8] [Ideas]
Hi, 

I'd like to have a card displaying my kodi playlist. After a couple of days looking for what I need, I had I can't find it and decided to program it by myself. I started from your custom-component to build a new sensor for the playlist and wrote my own card. Everything is working (card is just very ugly for the moment), but as I basically copied all the code directly in HA, I have no build process in place using HACS. Another point is that I duplicate all the code which is against the 'reusable' principle.

So... my question is: would you like to extend your project by adding more sensors (eventually customizable)? I can fork your project and propose changes, but I'll have to dig more into GIT to understand how this works. I can also show you what I changed to get the new sensor and the JsonRPC call if you want to. It would be cool to have also a new project for a new card, but as the original idea comes from you, maybe better if you create it?

Anyway, if interested, just let me know.

Gautier

**Comment by boralyl** [comment]
Gautier,

I"m glad you were able to modify this code to meet your needs for fetching playlists to display in HA!  This particular project is probably not the best place to add additional sensors as it's essentially a provider for the [Upcoming Media Card](https://github.com/custom-cards/upcoming-media-card).  As you know it's pretty simple and just does an rpc call to fetch new movies/tv shows.  The sensors store data in a very specific format so that the card referenced above can display it in HA.

I am probably also wouldn't be the best person to start a new project for a new card.  My front-end skills aren't very great, I'm much better at the python side of things :)

I encourage you to start a new repo forked off of my project, I'm sure you are not the only one who would find the playlist component/card useful!

Cheers,
Aaron

**Comment by jtbgroup** [comment]
Hi Aaron,

Thanks a lot for your reply!
I wanted to have a result of my investigation before answering your mail.
As you can see, I now have my playlist in HA with some controls.
[image: image.png]

I come from the java world, but didn't program for a while. So getting back
into Python and Javascript is not easy, but I learn everyday!

I forked your Custom-Component repo and added some entities. As I'll keep
the same philosophy, I can of course start a new repo and publish it, but
90% of the code comes from you and I don't want to "steal" it, I'd find
more honest, regarding the intellectual rights, to continue with your
project. But it's up to you; you probably have other projects ;-)

The card is a new one, based on the properties of the playlist.
I didn't search for the moment, but is it difficult to publish a component
via HACS? But of course, I can't publish the card without the
custom-component.

Regards,

Gautier


Le ven. 8 janv. 2021 à 17:30, Aaron Godfrey <notifications@github.com> a
écrit :

**Comment by boralyl** [comment]
Gautier,

I can't see the image you posted, but I"m glad you've been able to add some additional controls to it!  I don't think I've ever used playlists in kodi.

This repository is open source so anyone can use the code, no need to worry about stealing or taking credit for this code.  If you want to you can always give credit in a README.  I do have a lot of projects I work on in my free time outside of my public repos, so I don't have much bandwidth for taking on new projects.

The [HACS website](https://hacs.xyz/docs/publish/start) has some good documentation on how to publish your custom component or card.  I'd start there and read over their docs.

Cheers!
Aaron

**Comment by jtbgroup** [comment]
Hi,

I made another screen capture...

Understood for the repository. I'll make a new one and of course, mention
you in the readme!

Thanks for the link. I'll have a look when everything is ready.

I'll probably let you know when everything is ready to be published.

Gautier



Le dim. 10 janv. 2021 à 20:04, Aaron Godfrey <notifications@github.com> a
écrit :
